src/backend/distributed/executor/multi_utility.c

```diff
@@ -1583,7 +1610,6 @@ ErrorIfUnsupportedAlterTableStmt(AlterTableStmt *alterTableStatement)
 			case AT_DropColumn:
 			case AT_ColumnDefault:
 			case AT_AlterColumnType:
-			case AT_SetNotNull:
```

Comment:
I think other 4 constraints should be prevented to apply them over the partition column, but adding NOT NULL constraint over the partition column won't cause a problem.